File: backend/api/save_interaction.py
import json

def handler(request):
    if request.method != "POST":
        return {
            "statusCode": 405,
            "body": "Method Not Allowed"
        }

    try:
        body = json.loads(request.body)
        user_text = body.get("user_text", "").strip()

        if not user_text:
            return {
                "statusCode": 400,
                "body": json.dumps({"error": "Texto vacío"})
            }

        with open("/tmp/interacciones.txt", "a", encoding="utf-8") as f:
            f.write(user_text + "\n")

        return {
            "statusCode": 200,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"status": "ok", "received": user_text})
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "body": json.dumps({"error": str(e)})
        }

File: backend/api/save_interaction.js
// backend/api/save_interaction.js
import fs from "fs";
import path from "path";

export default async function handler(req, res) {
  if (req.method === "OPTIONS") {
    res.setHeader("Access-Control-Allow-Origin", process.env.FRONTEND_ORIGIN || "*");
    res.setHeader("Access-Control-Allow-Methods", "POST, OPTIONS");
    res.setHeader("Access-Control-Allow-Headers", "Content-Type");
    return res.status(200).end();
  }

  if (req.method !== "POST") return res.status(405).send("Method Not Allowed");

  try {
    const { user_text } = req.body || JSON.parse(req.rawBody || "{}");
    if (!user_text || !user_text.trim()) return res.status(400).json({ error: "Texto vacío" });

    const filePath = path.join("/tmp", "interacciones.txt");
    fs.appendFileSync(filePath, user_text + "\n", { encoding: "utf8" });

    res.setHeader("Access-Control-Allow-Origin", process.env.FRONTEND_ORIGIN || "*");
    return res.status(200).json({ status: "ok", received: user_text });
  } catch (e) {
    console.error("save_interaction error:", e);
    return res.status(500).json({ error: String(e) });
  }
}
